packages/fm-player/src/components/SoundBoard.tsx

```diff
@@ -5,7 +5,9 @@ import SoundButton from './SoundButton';
 import Search from './Search/Search';
 
 function soundMatchesPhrase(sound: XSound, phrase: string) {
-  return sound.name.includes(phrase) || (sound.tags || []).some((tag) => tag.includes(phrase));
+  const { name, tags } = sound;
+  const tagsAndNameJoinedString = [name, ...tags ?? []].join('').toLowerCase();
+  return phrase.toLowerCase().split(' ').every((keyword) => tagsAndNameJoinedString.includes(keyword));
```

Comment:
Just an observation: I was trying to come up with an idea that would not include concatenating strings, but `.includes()` right here makes it challening to replace it with a proper data structure that would be easily readable. Let's stick to what you came up with :).